packages/display_renderers/src/robot_info_renderer_node.py

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env python3
+
+import rospy
+import requests
+from typing import Dict
+from duckietown_msgs.msg import DisplayFragment as DisplayFragmentMsg
+
+from display_renderer import (
+    MonoImageFragmentRenderer,
+    PAGE_ROBOT_INFO,
+    REGION_BODY,
+    DisplayROI,
+)
+from display_renderer.text import monospace_screen
+
+from duckietown.dtros import DTROS, NodeType, TopicType
+
+
+class RobotInfoRendererNode(DTROS):
+    def __init__(self):
+        super(RobotInfoRendererNode, self).__init__(
+            node_name="robot_info_renderer_node", node_type=NodeType.VISUALIZATION
+        )
+        # get parameters
+        self._veh = rospy.get_param("~veh")
+        self._frequency = rospy.get_param("~frequency")
+        # create publisher
+        self._pub = rospy.Publisher(
+            "~fragments",
+            DisplayFragmentMsg,
+            queue_size=1,
+            dt_topic_type=TopicType.VISUALIZATION,
+            dt_help="Fragments to display on the display",
+        )
+        # create renderers
+        self._renderer = RobotInfoRenderer()
+        # create loop
+        self._timer = rospy.Timer(rospy.Duration.from_sec(1.0 / self._frequency), self._beat)
+
+    def _beat(self, _):
+        robot_info_str = self._fetch()
+        self._renderer.set_contents(robot_info_str)
+        self._pub.publish(self._renderer.as_msg())
+
+    def _fetch(self) -> str:
+        health_api_url = f"http://{self._veh}.local/health/"
+        # noinspection PyBroadException
+        try:
+            health_data = requests.get(health_api_url).json()
+        except BaseException:
+            return
+
+        sw_date = health_data["software"]["date"]
+        sw_version = health_data["software"]["version"]
+        firmware = f"{sw_date['day']}/{sw_date['month']}/{sw_date['year']} ({sw_version})"
+
+        return f"Name: {self._veh}\nModel: {health_data['hardware']['model']}\nFirmware:\n{firmware}"
```

Comment:
Use the function `get_robot_configuration()` (defined here: https://github.com/duckietown/dt-commons/blob/4046177f07d3ba743f89d769ab7323c68154df47/packages/dt_robot_utils/robot.py) to get the robot model.

Reply:
Let's justify the text, align the key to the left side of the screen and the value to the right.

Reply:
The texts are a bit too small to read, when the firmware version is on the same line. Here's what it looks like:

![image](https://github.com/duckietown/dt-duckiebot-interface/assets/10885835/4293e7dc-a0e1-479e-accc-068b1224df1b)